### PR TITLE
Generics tasks

### PR DESCRIPTION
**Pull request by a-likhtar**

### CODE REVIEW

File: 02-Generics/Generics/Generics.cs [hunk 1]
```diff
@@ -23,8 +29,7 @@ public static class ListConverter {
 		///   { new TimeSpan(1, 0, 0), new TimeSpan(0, 0, 30) } => "01:00:00,00:00:30",
 		/// </example>
 		public static string ConvertToString<T>(this IEnumerable<T> list) {
-			// TODO : Implement ConvertToString<T>
-			throw new NotImplementedException();
+            return string.Join(ListSeparator.ToString(), list.Select(value => value));
```
Comment (AlexMokin) [inline 02-Generics/Generics/Generics.cs]: do you need `Select` here?

File: 02-Generics/Generics/Generics.cs [hunk 2]
```diff
@@ -60,42 +73,75 @@ public static class ArrayExtentions {
...
 		/// <param name="index1">first index</param>
 		/// <param name="index2">second index</param>
-		public static void SwapArrayElements<T>(this T[] array, int index1, int index2) {
-			// TODO : Implement SwapArrayElements<T>
-			throw new NotImplementedException();
-		}
+		public static void SwapArrayElements<T>(this T[] array, int index1, int index2)
+		{
+		    
+		    SwapElement(ref array[index1], ref array[index2]);
```
Comment (AlexMokin) [inline 02-Generics/Generics/Generics.cs]: You don't need additional function here

File: 02-Generics/Generics/Generics.cs [hunk 3]
```diff
@@ -105,11 +151,15 @@ public static class ArrayExtentions {
...
+	public static class Singleton<T>
+        where T:new()
+    {
 		// TODO : Implement generic singleton class 
 
+	    private static readonly T _Instance = new T();
+
 		public static T Instance {
-			get { throw new NotImplementedException(); }
+			get { return _Instance; }
```
Comment (AlexMokin) [inline 02-Generics/Generics/Generics.cs]: what is the proper name for static fields?

File: 02-Generics/Generics/Generics.cs [hunk 4]
```diff
@@ -134,9 +184,35 @@ public static class FunctionExtentions {
...
+		            return result;
+		        }
+		        catch (WebException ex)
+		        {
+		            System.Diagnostics.Trace.WriteLine(ex);
+		            count++;
+		        }
+		    }
+
+		    try
```
Comment (AlexMokin) [inline 02-Generics/Generics/Generics.cs]: you can do it without 2nd try catch block

**Comment by AlexMokin** [comment]
well done. do the next task

**Comment by AlexMokin** [comment]
well done. Do the next task

**Comment by IlyaMokin** [comment]
closed

**Pull request closed without merging**